### PR TITLE
[SPARK-25859][ML]add scala/java/python example and doc for PrefixSpan

### PR DESCRIPTION
**Pull request by huaxingao**
## What changes were proposed in this pull request?

add scala/java/python example and doc for PrefixSpan in branch 2.4

## How was this patch tested?

Manually tested

### CODE REVIEW

**Comment by huaxingao** [comment]
@felixcheung

File: examples/src/main/scala/org/apache/spark/examples/ml/PrefixSpanExample.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      .setMaxPatternLength(5)
+      .setMaxLocalProjDBSize(32000000)
+      .findFrequentSequentialPatterns(df)
+      .show()
+    // $example off$
+
+    spark.stop()
+  }
+}
+// scalastyle:on println
```
Comment (felixcheung) [inline examples/src/main/scala/org/apache/spark/examples/ml/PrefixSpanExample.scala]: nit: looks like println is not used in example here

**Review by felixcheung — APPROVED** [review]

**Comment by felixcheung** [comment]
merged to 2.4

**Comment by felixcheung** [comment]
please close this PR. thx

**Pull request closed without merging**